### ISSUE TITLE
Add autocomplete attribute configuration in form builder

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail#13629]
### Is your proposal related to a problem?

Forms created with Wagtail’s form builder can’t leverage the [autocomplete](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Attributes/autocomplete) attribute. This kind of automated assistance really helps with speeding up data entry, and having that available would help site implementers meet [WCAG SC 3.3.7 Redundant Entry](https://www.w3.org/TR/WCAG22/#redundant-entry) when they have forms with the same info to be filled multiple times.

### Describe the solution you'd like

Add a way to set [autocomplete](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Attributes/autocomplete) values per field.

### Describe alternatives you've considered

See [New StreamField-based form builder #11389](https://github.com/wagtail/wagtail/discussions/11389)

### Additional context

I’m not sure how this would work as in Django this attribute is dependent on the widget in use for each form field? See #6132

### Working on this

Tagging **Needs Design Decision**. For now I don’t think we can progress this until we’ve discussed [New StreamField-based form builder #11389](https://github.com/wagtail/wagtail/discussions/11389) more. This would be much simpler and nicer to implement within a new form builder paradigm.

Contribute to the discussion to help this move forward. Explain your use cases, discuss possible solutions, research how to adapt existing code, or compatibility concerns.

Once others agree with the proposed changes, anyone can contribute to this. View our [contributing guidelines](https://docs.wagtail.org/en/latest/contributing/index.html). Add a comment to the issue to explain what approach you are taking.

**Comment by dolliecoder** [comment]
I’d like to work on this issue 🙂

**Comment by thibaudcolas** [comment]
@dolliecoder see above:

> For now I don’t think we can progress this until we’ve discussed https://github.com/wagtail/wagtail/discussions/11389 more. This would be much simpler and nicer to implement within a new form builder paradigm.

There is nothing to work on here yet, aside from contributing to that discussion.

**Comment by dolliecoder** [comment]
> [@dolliecoder](https://github.com/dolliecoder) see above:
> 
> > For now I don’t think we can progress this until we’ve discussed [#11389](https://github.com/wagtail/wagtail/discussions/11389) more. This would be much simpler and nicer to implement within a new form builder paradigm.
> 
> There is nothing to work on here yet, aside from contributing to that discussion.

Thanks for the clarification, @thibaudcolas! I’ll check out #11389 and join the discussion there. Appreciate the guidance — looking forward to contributing once things move forward!

**Comment by Luzzzi** [comment]
This will be very useful to us in our situation. A recent accessibility audit highlighted the lack of autocomplete functionality in our forms. Is there anything we can do to help?

**Comment by thibaudcolas** [comment]
@Luzzzi this is currently earmarked for implementation on top of a large refactoring, switching to a [StreamField-based form builder](https://github.com/wagtail/wagtail/discussions/11389). That’s a large amount of work so it’s [earmarked on our roadmap](https://github.com/wagtail/roadmap/issues/120). Once that’s done, adding the support for the attribute should be very simple.

The alternative would be to add support with the current implementation. I don’t think that’s a good fit for us as our current model doesn’t support conditional fields (#2172), but if someone implements that first we could revisit. Technically we already have a `RulesController` that supports conditional fields, but we’re missing a neat enough API to do this in scenarios like this, particularly within InlinePanel with field names prefixed with item identifiers.
